mllib/src/test/scala/org/apache/spark/ml/clustering/KMeansSuite.scala

```diff
@@ -152,6 +158,35 @@ class KMeansSuite extends SparkFunSuite with MLlibTestSparkContext with DefaultR
     val kmeans = new KMeans()
     testEstimatorAndModelReadWrite(kmeans, dataset, KMeansSuite.allParamSettings, checkModelData)
   }
+
+  test("training with initial model") {
+    val kmeans = new KMeans().setK(2).setSeed(1)
+    val model1 = kmeans.fit(rData)
+    val model2 = kmeans.setInitMode("initialModel").setInitialModel(model1).fit(rData)
+    model2.clusterCenters.zip(model1.clusterCenters)
+      .foreach { case (center2, center1) => assert(center2 ~== center1 absTol 1E-8) }
+  }
+
+  test("training with initial model, error cases") {
+    val kmeans = new KMeans().setK(k).setSeed(1).setMaxIter(1)
+
+    // Sets initMode with 'initialModel', but does not specify initial model.
+    intercept[IllegalArgumentException] {
```

Comment:
@sethah +1 on the behavior you purpose. The only thing I would like to add on is `setK` should throw `IllegalArgumentException`.

```scala
    // initialModel sets k and init mode
    assert(km.getInitMode === MLlibKMeans.K_MEANS_INITIAL_MODEL)
    assert(km.getK === initialK)
    assert(km.getInitialModel.getK === initialK)

    // setting k will throw exception.
    intercept[IllegalArgumentException] {
      km.setK(initialK + 1)
    }
```